### ISSUE TITLE
Staking with simple Token 

### ISSUE THREAD

**Issue by maxmehlbaum** [OpenSTFoundation/openst-platform#93]
Do we have to deposit simple token into the reserve address in order to mint new tokens? is this how they are staked? also if so is this why the terminal is stuck at:

InterComm Script for Stake and Mint initiated 

both of our computers are stuck at this stage.

is it because its waiting to have the coins be deposited so they can then be recorded on the value chain. Also if so is this why its called the value chain because it is where you stake the value backing our coin. the backing being simple token?

**Comment by maxmehlbaum** [comment]
we ended up trying to Start minting after a couple hours of waiting for the branded tokens to be staked by OST
  > cd openst-platform
  > source test/open_st_env_vars.sh
  > node tools/stake_and_mint/for_branded_token.js
  Note: ACME key password is "testtest"


we entered 1 for Please choose member to fund. 
1 ' for ' 'Acme Coin' '(' 'ACME' ')'
2 ' for ' undefined '(' 'ACME' ')'
3 ' for ' undefined '(' 'ACME' ')'
4 ' for ' undefined '(' 'ACME' ')'
>1
Confirm Member 
 Please Confirm these details. (we confirmed)
>yes

Please mention the Simple Tokens to Assign.
>1

we put 1 as a test. is this 1 the amount of simple token that is backing our BT??


this was found in the middle of the terminal processes:
**** No contract found for contract address: (for our address)
but we received this at the end
Yoo.. Have Fun!!  

so we then went to the next command of opening up a new terminal 

>npm start 




we received:


> openst-platform@0.0.0 start /Users/max/Downloads/openst-platform-master
> node ./bin/www

path.js:28
    throw new TypeError('Path must be a string. Received ' + inspect(path));
    ^

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.isAbsolute (path.js:1235:5)
    at Object.<anonymous> (/Users/max/Downloads/openst-platform-master/config/core_constants.js:60:11)
    at Module._compile (module.js:643:30)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! openst-platform@0.0.0 start: `node ./bin/www`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the openst-platform@0.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/max/.npm/_logs/2018-01-23T05_02_27_430Z-debug.log

this is where we are currently. Please let us know if we were impatient or missed a step.

**Comment by jasonklein** [comment]
Working from the bottom up:

>path.js:28
throw new TypeError('Path must be a string. Received ' + inspect(path));
^

This error is likely due to running `npm start` in a new terminal without having reloaded the environment variables beforehand (`source test/open_st_env_vars.sh`).

In the stack trace provided, there is:

```
TypeError: Path must be a string. Received undefined
at assertPath (path.js:28:11)
at Object.isAbsolute (path.js:1235:5)
at Object. (/Users/max/Downloads/openst-platform-master/config/core_constants.js:60:11)
```

At/around `config/core_constants.js:60:11`:

```
// file path of the member config file.
var configFilePath = process.env.OST_MEMBER_CONFIG_FILE_PATH;
if (!path.isAbsolute(configFilePath)) {
  configFilePath = path.join(__dirname, '/' + rootPrefix + '/' + configFilePath);
}
```

The stack trace indicates that `configFilePath` is undefined, but L59 in `config/core_constants.js` defines `configFilePath` as `process.env.OST_MEMBER_CONFIG_FILE_PATH`. So, for some reason, that definition is not reaching the app.

---
> **** No contract found for contract address: (for our address)

This output comes from https://github.com/OpenSTFoundation/openst-platform/blob/master/lib/web3/events/decoder.js#L69

The error message is somewhat unclear, as these tools were built for internal development. This error indicates that a matching contract ABI cannot be found in order to decode an event log, not that the contract does not exist. For example, if ABIs are known for contracts A and C, a transaction is executed that calls A --> B --> C, and B emits an event, this error informs the developer that the log  of that event from B in the transaction hash could not be decoded because the ABI for B is not known.

Note that the full error might look something like:

`**** No contract found for contract address: 0x86D59e4271e8591663654f0838ccF088ba91e61E at log index(0) in TxHash: 0x28576c97b66947ebdb6b83d4e094165aa5df8936ad8ab16857ba623db60fdc44`

---
> we put 1 as a test. is this 1 the amount of simple token that is backing our BT??

Yes, this is the amount of Simple Token elected to be staked to mint branded tokens.

During the execution of `./run_complete_test_deploy.sh`, a branded token, "Acme Coin" was deployed on the utility chain and registered on the value chain; and the staker address for that token was given 100 Simple Token of which some or all can be staked to mint Acme Coin.

---
>we entered 1 for Please choose member to fund.
1 ' for ' 'Acme Coin' '(' 'ACME' ')'
2 ' for ' undefined '(' 'ACME' ')'
3 ' for ' undefined '(' 'ACME' ')'
4 ' for ' undefined '(' 'ACME' ')'

Options 2-4 are listed because each time this process is re-run (without completely chucking everything), an additional Acme member is added to the members object in the config; but for the chains that are likely running, only option 1 is actually available (because the other options are for chains that have been abandoned). It is expected that this bug will be corrected in the next release (see #94).

---
>InterComm Script for Stake and Mint initiated
>
>both of our computers are stuck at this stage.

For the sake of completion/posterity (because you already sorted this out), I will add that after starting `node services/inter_comm/stake_and_mint.js`, the developer is meant to start `node tools/stake_and_mint/for_branded_token.js` in a different terminal in order to stake and mint. The intercomm is a service that acts upon instruction—so it was not stuck, but waiting. It is unfortunate you were held up a long while/thought something was wrong; we will clarify this in the README.

---
>Do we have to deposit simple token into the reserve address in order to mint new tokens? is this how they are staked?

Simple Tokens must be staked in order to mint branded tokens. As noted above, if the `./run_complete_test_deploy.sh` was successfully executed, then 100 Simple Tokens were transferred to the address used by `node tools/stake_and_mint/for_branded_token.js` to mint branded tokens.

Let us know if this gets you moving again, @maxmehlbaum!
